frappe/utils/boilerplate.py

```diff
@@ -26,8 +26,8 @@ def make_boilerplate(dest, app_name):
 		hook_val = None
 		while not hook_val:
 			hook_val = cstr(raw_input(key + ": "))
-			if hook_key=="app_name" and hook_val.lower().replace(" ", "_") != hook_val:
-				print "App Name must be all lowercase and without spaces"
+			if hook_key=="app_title" and not re.match("^[A-Za-z ]*$", hook_val):
```

Comment:
Use the same validation as doctype.py's `validate_name` i.e. python module name can have letters, numbers and underscores, but it should not start with underscore or a number. Hence, the App Title should have a similar validation.